From ac4cdc0cb199df357f4dadaa4fe7de04ec3b5605 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:25 -0400
Subject: [acpi] hotplug: Add .fixup() callback to struct acpi_hotplug_context

Message-id: <1411129776-21656-51-git-send-email-prarit@redhat.com>
Patchwork-id: 96386
O-Subject: [RHEL7.1 PATCH BZ 1141846 50/61] ACPI / hotplug: Add .fixup() callback to struct acpi_hotplug_context
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit 59b42fa01fe2d84f3c3f28ee6f25510820ace35b
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Feb 21 01:08:51 2014 +0100

    ACPI / hotplug: Add .fixup() callback to struct acpi_hotplug_context

    In order for the ACPI dock station code to be able to use the
    callbacks pointed to by the ACPI device objects' hotplug contexts
    add a .fixup() callback pointer to struct acpi_hotplug_context.
    That callback will be useful to handle PCI devices located in
    dock stations.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Stefan Assmann <sassmann@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index ff8af5e..b77175e 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -80,7 +80,7 @@ static struct acpiphp_context *acpiphp_init_context(struct acpi_device *adev)
   return NULL;
 
  context->refcount = 1;
- acpi_set_hp_context(adev, &context->hp, acpiphp_hotplug_event);
+ acpi_set_hp_context(adev, &context->hp, acpiphp_hotplug_event, NULL);
  return context;
 }
 
diff --git a/include/acpi/acpi_bus.h b/include/acpi/acpi_bus.h
index 989538c..18f204c 100644
--- a/include/acpi/acpi_bus.h
+++ b/include/acpi/acpi_bus.h
@@ -120,6 +120,7 @@ struct acpi_scan_handler {
 struct acpi_hotplug_context {
  struct acpi_device *self;
  int (*event)(struct acpi_device *, u32);
+ void (*fixup)(struct acpi_device *);
 };
 
 /*
@@ -340,10 +341,12 @@ static inline void acpi_set_device_status(struct acpi_device *adev, u32 sta)
 
 static inline void acpi_set_hp_context(struct acpi_device *adev,
            struct acpi_hotplug_context *hp,
-           int (*event)(struct acpi_device *, u32))
+           int (*event)(struct acpi_device *, u32),
+           void (*fixup)(struct acpi_device *))
 {
  hp->self = adev;
  hp->event = event;
+ hp->fixup = fixup;
  adev->hp = hp;
 }
 
-- 
1.7.1